flamedisx/lxe_blocks/quanta_generation.py

```diff
@@ -149,8 +149,8 @@ def _compute(self,
         work = self.gimme('work', data_tensor=data_tensor, ptensor=ptensor)
         mean_q_produced = (
                 energy_noStep
-                * self.gimme('lindhard_l', bonus_arg=energy_noStep,
-                             data_tensor=data_tensor, ptensor=ptensor)
+                * self.gimme('lindhard_l', bonus_arg=energy_noStep[:, 0, :],
```

Comment:
Can you explain what is happening here? What are the explicit [:, 0, :]'s doing / why do we need them now and not before?

Reply:
This was just to speed things up: before, `lindhard_l` was evaluated over a tensor with dimensions (energy_spectrum_length, nq_max - nq_min, events). But the function only depends on energy, not nq. So the computation was being repeated unnecessarily, and it was quite time consuming too, which meant the default NR models scaled quite poorly for longer energy spectra. This evaluates the function just over the dimensions (energy_spectrum_length, events), and then repeats it over nq.

Reply:
Thanks, that's a good optimization!